### PR TITLE
ci: add clang-12 to testing

### PR DESCRIPTION
**Pull request by szb512**
The title explains itself, adds clang-12 to testing.

{also I do not bother naming my branches anymore}

### CODE REVIEW

File: .github/workflows/build_test.yml [hunk 1]
```diff
@@ -21,6 +21,7 @@ jobs:
           - { COMPILER: "gcc",   COMPILER_VERSION: "10" }
           - { COMPILER: "clang", COMPILER_VERSION: "10" }
           - { COMPILER: "clang", COMPILER_VERSION: "11" }
+          - { COMPLIER: "clang", COMPILER_VERSION: "12" }
```
Comment (bluca) [inline .github/workflows/build_test.yml]: s/COMPLIER/COMPILER/

Comment (bluca) [reply .github/workflows/build_test.yml]: although I do hope it complies too :-)

**Comment by szb512** [comment]
Travis failure is unrelated.
toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit

The command "$CI_MANAGERS/debian.sh SETUP" failed and exited with 1 during .

Your build has been stopped.

**Comment by szb512** [comment]
github's test says something about a line 59, but I do not know what file it is talking about.

**Comment by bluca** [comment]
> github's test says something about a line 59, but I do not know what file it is talking about.

There's a typo in the patch, I've highlighted it above

**Comment by szb512** [comment]
I see the highlighted line, but I do not see the typo.

**Comment by poettering** [comment]
> I see the highlighted line, but I do not see the typo.

COMPLIER → COMPILER

L and I are switched

**Comment by szb512** [comment]
ohhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh
ok

**Comment by szb512** [comment]
Welp the typo fix still caused clang-12 to fail

**Comment by szb512** [comment]
E: The repository 'http://apt.llvm.org/focal llvm-toolchain-focal-12 Release' does not have a Release file.
Error: Process completed with exit code 100.

**Comment by mrc0mmand** [comment]
That's because there's no branch for the clang 12, yet. It's only in the snaphsot/development branch.

Either we wait until clang 12 is branched out or tweak the build script to use the snapshot branch under certain conditions. My preference would be to do the former :-)

**Comment by bluca** [comment]
> That's because there's no branch for the clang 12, yet. It's only in the snaphsot/development branch.
> 
> Either we wait until clang 12 is branched out or tweak the build script to use the snapshot branch under certain conditions. My preference would be to do the former :-)

Let's wait for the full release and availability please. We got enough flaky and unrelated failings in our PRs to deal with already :-)
Closing for now, please send again once that happens.

**Pull request closed without merging**